lib/Normalize_std_ast.ml

```diff
@@ -132,10 +172,31 @@ let make_mapper conf ~ignore_doc_comments ~erase_jane_syntax =
           (Exp.sequence ~loc:loc1 ~attrs:attrs1
              (Exp.sequence ~loc:loc2 ~attrs:attrs2 exp1 exp2)
              exp3 )
+    | Pexp_apply
+        ( {pexp_desc= Pexp_extension ({txt= extension_name; _}, PStr []); _}
+        , [(Nolabel, sbody)] )
+      when Conf.is_jane_street_local_annotation "local" ~test:extension_name
+           || Conf.is_jane_street_local_annotation "exclave"
+                ~test:extension_name ->
+        let is_fun =
+          match sbody.pexp_desc with Pexp_fun _ -> true | _ -> false
+        in
+        m.expr m
+          { sbody with
+            pexp_attributes=
+              convert_legacy_jane_street_local_annotations ~ghost:is_fun
```

Comment:
This condition turns out to not be fully correct. The only case where we want location ghost here is if **both** `is_fun` is true and it's a value binding inside of a let. Something like:

```
let g = [%local] (fun a b c -> 1)
```

The reason I imagine is because it gets rewritten to:

```
let local_ g a b c = 1
```

where the `fun` keyword is no longer present.

I pushed a new commit to fix this which will handle cases like `let g = f ([%local] (fun a b c -> 1))` correctly.